### PR TITLE
Add H2 upgrade to release notes and migration guide

### PR DESCRIPTION
**Pull request by ahus1**
Relates to keycloak/keycloak#12607
Relates to keycloak/keycloak#11292
Relates to keycloak/keycloak#10113

### CODE REVIEW

**Comment by ahus1** [comment]
@hmlnarik - as you've been involved with the original PR keycloak/keycloak#14819, I ask you to review this, or involve another developer.

@andymunro - could you please be so kind to review the contents that they comply with the style guide. 

Thanks!

File: release_notes/topics/20_0_0.adoc [hunk 1]
```diff
@@ -9,4 +9,11 @@ More details can be found at the https://www.keycloak.org/server/hostname[Config
 
 = Improvements to `kc.bat` when running Keycloak on Windows
 
-In this release, we are making important changes to `kc.bat` in order to give the same experience as when running on Linux.
\ No newline at end of file
+In this release, we are making important changes to `kc.bat` in order to give the same experience as when running on Linux.
```
Comment (andymunro) [inline release_notes/topics/20_0_0.adoc]:
```suggestion
In this release, we are making important changes to `kc.bat` to give the same experience as when running on Linux.
```

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 2]
```diff
@@ -1,3 +1,30 @@
...
\ No newline at end of file
+Updated the RESTEasy version of Keycloak Admin REST Client to the next major version.
+
+= H2 Version Update
+
+{project_name} ships for development purposes with an H2 database driver.
+As it is intended for development purposes only, it should never be used in a production environment.
+
+In this release, the H2 driver has been upgraded from version 1.x to version 2.x.
+This might require changes to the H2 JDBC URL or migration of the H2 database files in an existing Keycloak setup.
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]:
```suggestion
This change might require changes to the H2 JDBC URL or migration of the H2 database files in an existing Keycloak setup.
```

File: upgrading/topics/keycloak/changes-20_0_0.adoc [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+
+== Changes to the H2 JDBC URL
+
+For Keycloak to run with its JPA legacy store with H2 version 2.x, the JDBC URL requires the attribute `NON_KEYWORDS=VALUE`.
+
+In a setup where H2 is initialized by Keycloak without extra parameters, Keycloak will append the attribute automatically.
+This is the default for development setups.
+
+If an H2 JDBC URL is provided on the command line or a configuration file, and it already contains the `NON_KEYWORDS=` attribute in the JDBC URL, this attribute needs to be amended with the `VALUE` keyword.
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes-20_0_0.adoc]:
```suggestion
If an H2 JDBC URL is provided on the command line or in a configuration file, and it already contains the `NON_KEYWORDS=` attribute in the JDBC URL, this attribute needs to be amended with the `VALUE` keyword.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good, @ahus1. Just a few suggestions.

**Comment by ahus1** [comment]
@andymunro - thanks for the review, I applied the changes. Will ask you for a re-review once I received a first feedback from @hmlnarik

**Review by andymunro — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]

**Pull request merged**